### PR TITLE
Dispatch to the main thread first and create the alert

### PR DESCRIPTION
**Pull request by RPangrle**
The associated object call created some memory management issues when the server is misbehaving.

### CODE REVIEW

File: ADAL/src/ui/ios/ADNTLMUIPrompt.m [hunk 1]
```diff
@@ -21,26 +21,60 @@
...
+        UITextField* username = alert.textFields.firstObject;
+        UITextField* password = alert.textFields.lastObject;
+        
+        block(username.text, password.text);
     }];
+    
+    [alert addAction:cancelAction];
+    [alert addAction:loginAction];
+    
+    [alert addTextFieldWithConfigurationHandler:^(UITextField * _Nonnull textField) { (void)textField; }];
```
Comment (jasoncoolmax) [inline ADAL/src/ui/ios/ADNTLMUIPrompt.m]: Shall we add a placeholder like "Username" and "Password"? Currently it seems empty for both textboxes :)

Comment (RPangrle) [inline ADAL/src/ui/ios/ADNTLMUIPrompt.m]: MacOS UI has labels on it, placeholders are less common.

Comment (jasoncoolmax) [inline ADAL/src/ui/ios/ADNTLMUIPrompt.m]: OK :)

**Comment by jasoncoolmax** [comment]
:shipit:

File: ADAL/src/ui/ios/ADNTLMUIPrompt.m (hunk 1 above)
Comment (RPangrle) [inline ADAL/src/ui/ios/ADNTLMUIPrompt.m]: Oh wait, sorry, this is iOS UI. The Safari version of this control doesn't user placeholders and then we would have to localize those strings, so it's not worth it.

**Comment by jasoncoolmax** [comment]
:shipit:

**Pull request merged**